### ISSUE TITLE
Unable to find packages.

### ISSUE THREAD

**Issue by pnunn** [mdfaizan7/go-authentication-boilerplate#1]
Faizan, 

firstly, thank you for the enormous effort in putting together this extensive tutorial, it is fantastic in its detail and coverage.

I have coded up the whole thing, and I've hit, what I know is a newbie issue and is nothing to do with your code.

I can't get it to run. I'm also sure its something to do with my not understanding go packages and modules.

When I try 'go run main.go' I'm getting

`
go run main.go                                                                                12:49:43main.go:6:2: package go-authenticate-boilerplate/database is not in GOROOT (/usr/local/go/src/go-authenticate-boilerplate/database)
main.go:7:2: package go-authenticate-boilerplate/router is not in GOROOT (/usr/local/go/src/go-authenticate-boilerplate/router)
`
which is technically correct, the go-authenticate-boilerplate is not in those paths.

When I do 'go env' I get this back

`
go env                                                                                        12:49:52GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/pnunn/.cache/go-build"
GOENV="/home/pnunn/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/pnunn/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/pnunn/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/pnunn/go/src/github.com/pnunn/go-authentication-boilerplate/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build251484057=/tmp/go-build -gno-record-gcc-switches"
`
Which actually seems to be correct in terms of the GOPATH and GOMOD. What am I doing wrong? 

Thanks heaps.
Peter.

**Comment by mdfaizan7** [comment]
Hey @pnunn, I'm glad that you liked the tutorial!

This is a very common issue with Golang while setting it up. I think that you are facing this issue because you have not turned on Go Modules. You need to turn on your GOModule in your bashrc file: 

```bash
export GO111MODULE=on
```

Also, make sure to check your GOPATH in environment variables:
```bash
GO111MODULE="on"
GOPATH="/xyz/gopath"
```

**Comment by mdfaizan7** [comment]
To be

**Comment by pnunn** [comment]
Thanks for the input.

I'm sure this is a problem with my setup not you code.. :+1: 

I have my code under /home/pnunn/go/src/github.com/pnunn/go-authentication-boilerplate

I exported the GO111MODULE variable but that didn't seem to help.

I added github.com/pnunn/ to the front of the imports for the boilerplate imports, but that now tries to download them from git (and there local).

I'm really missing something here, it can't be this hard surely.

An one point, I did manage to get the third party packages to install by the looks of things (I think I set the GOPATH to be /home/pnunn/go/src/github.com/pnunn to get that working) but the local boilerplate ones still didn't install. I think I'm close. Is there a definitive file layout that I should use? On linux I thought it was to use /home/usr/go/src/github.com/user/.... but that seems to be an issue.
Sorry if this is a basic question, but its got me stumped.

I seem to be able to use other code to run without the go.mod file if I use the full path to the imports.. So confused.
